File: 19주차/이연상/카펫.py
# [PRG] 카펫

def solution(brown, yellow):
    answer = []
    
    val = brown + yellow
    for i in range(1, val):
        if (val % i == 0):
            if(val / i) <= i:
                if 2*i + 2*(val / i) == brown + 4:
                    answer = [i, val / i]
                    break
    
    return answer

File: 40주차/이연상/카펫.java
// [PRG] 카펫 

class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        
        int val = brown + yellow;
        for(int i = 1; i < val; i++) {
            if(val % i == 0) {
                if((val / i) <= i) {
                    if(i *2 + (val / i) * 2 == brown+4) {
                        answer[0] = i;
                        answer[1] = val / i;
                    }  
                }
            }
        }
        
        return answer;
    }
}